pkg/deploy-manager/subscription.go

```diff
@@ -360,3 +519,83 @@ func (t *DeployManager) waitForOCSOperator() error {
 
 	return nil
 }
+
+func (t *DeployManager) updateClusterObjects(co *clusterObjects) error {
+	for _, catalogSource := range co.catalogSources {
+		cs, err := t.olmClient.OperatorsV1alpha1().CatalogSources(catalogSource.Namespace).Get(catalogSource.Name, metav1.GetOptions{})
+		cs.Spec.Image = catalogSource.Spec.Image
+		_, err = t.olmClient.OperatorsV1alpha1().CatalogSources(catalogSource.Namespace).Update(cs)
+		if err != nil {
+			return err
+		}
+
+	}
+
+	// Give some time for the catalogsource to be created
+	time.Sleep(120 * time.Second)
```

Comment:
is there a reason why we can't post the catalogSource and subscription changes at the same time without this sleep?

Reply:
This was to ensure that the catalog source is ready before updating the subscription. If olm does't detect the catalogsource to be ready within some time, I've noticed it backs off for over 10 minutes. I will explicitly check for the catalog-source to be ready here.

Reply:
ah yes. i've seen the same thing with the backoff. It's really frustrating. The wait until logic should at least help a little to make sure we're posting the subscription as soon as possible.